src/Runner.Worker/Handlers/ScriptHandler.cs

```diff
@@ -58,12 +58,19 @@ public override void PrintActionDetails(ActionRunStage stage)
             string shellCommandPath = null;
             bool validateShellOnHost = !(StepHost is ContainerStepHost);
             string prependPath = string.Join(Path.PathSeparator.ToString(), ExecutionContext.PrependPath.Reverse<string>());
-            Inputs.TryGetValue("shell", out var shell);
+            string shell = null;
```

Comment:
different cases to think through. Not sure what behavior makes sense:

- Default defined, shell input defined but empty.

Does this fallback to job default? Or does it fallback to the handler default?

Reply:
Consider from a template, how do you specify use the handler default.

Reply:
Since we dont have templates yet, should we add a TODO to figure out?

Reply:
Do we know if we're within a scope. Can we fail if inside template? That way we are forced to reconcile later rather than accidentally choosing a behavior :)